rasterio/rio/info.py

```diff
@@ -50,6 +50,8 @@
 @click.option('--checksum', 'meta_member', flag_value='checksum',
               help="Print integer checksum of a single band "
                    "(use --bidx).")
+@click.option('--subdatasets', 'meta_member', flag_value='subdatasets',
```

Comment:
I wonder if it wouldn't be more useful for this flag to print out a list of subdataset connection strings that are ready to be fed back into a `$ rio` command?  Something more like:

```console
$ rio info --subdatasets tests/data/RGB.nc
netcdf:/Users/sean/code/rasterio/tests/data/RGB.nc:Band1
netcdf:/Users/sean/code/rasterio/tests/data/RGB.nc:Band2
netcdf:/Users/sean/code/rasterio/tests/data/RGB.nc:Band3
```

I'm not sure what `GDALGetDescription()` reports for a NetCDF file, but if it is the same as what `src.subdatasets` is accessing then a `--description` flag on `$ rio info` might be a better option.